tools/nfsdist.py

```diff
@@ -137,16 +140,21 @@
 # common file functions
 b.attach_kprobe(event="nfs_file_read", fn_name="trace_entry")
 b.attach_kprobe(event="nfs_file_write", fn_name="trace_entry")
-b.attach_kprobe(event="nfs4_file_open", fn_name="trace_entry")
 b.attach_kprobe(event="nfs_file_open", fn_name="trace_entry")
 b.attach_kprobe(event="nfs_getattr", fn_name="trace_entry")
 
 b.attach_kretprobe(event="nfs_file_read", fn_name="trace_read_return")
 b.attach_kretprobe(event="nfs_file_write", fn_name="trace_write_return")
-b.attach_kretprobe(event="nfs4_file_open", fn_name="trace_open_return")
 b.attach_kretprobe(event="nfs_file_open", fn_name="trace_open_return")
 b.attach_kretprobe(event="nfs_getattr", fn_name="trace_getattr_return")
 
+if args.nfsvers == 4:
+    b.attach_kprobe(event="nfs4_file_open", fn_name="trace_entry")
+    b.attach_kretprobe(event="nfs4_file_open", fn_name="trace_open_return")
+elif args.nfsvers == 3:
+    b.attach_kprobe(event="nfs_file_open", fn_name="trace_entry")
```

Comment:
Is it possible to automatically detect the availability certain symbols to auto decide the nfs version, so we do not need this option?

Reply:
I'm not sure how to do that.  Do you have any suggestions?